timeutil/synchronized.go

```diff
@@ -0,0 +1,67 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package timeutil
+
+import (
+	"fmt"
+
+	"github.com/godbus/dbus"
+
+	"github.com/snapcore/snapd/dbusutil"
+	"github.com/snapcore/snapd/logger"
+)
+
+func isNoServiceOrUnknownPropertyDbusErr(err error) bool {
+	derr, ok := err.(dbus.Error)
+	if !ok {
+		return false
+	}
+	switch derr.Name {
+	case "org.freedesktop.DBus.Error.ServiceUnknown", "org.freedesktop.DBus.Error.UnknownProperty":
+		return true
+	}
+	return false
+}
+
+// IsNTPSynchronized returns true if the time is syncronized using something like
+// NTP.
+func IsNTPSynchronized() (bool, error) {
+	// shared connection, no need to close
+	conn, err := dbusutil.SystemBus()
+	if err != nil {
+		return false, fmt.Errorf("cannot connect to system bus: %v", err)
+	}
+
+	tdObj := conn.Object("org.freedesktop.timedate1", "/org/freedesktop/timedate1")
+	dbusV, err := tdObj.GetProperty("org.freedesktop.timedate1.NTPSynchronized")
+	if err != nil {
+		if isNoServiceOrUnknownPropertyDbusErr(err) {
```

Comment:
the logic here is that if i.e. there is nothing listening on the other end that this will never succeed and thus we should inform callers that the time has been synchronized because it either never will be or it will be synchronized by something else we aren't checking for? 

if so a comment would that this is the reasoning would be appreciated